### ISSUE TITLE
File attachment size

### ISSUE THREAD

**Issue by mmallejac** [timlegge/docker-foswiki#33]
Hello Tim,

Hope you're doing good.
I've noticed that attachment size is limited to 1M, most probably the default value for Nginx. Is that on purpose ?  I would set it to 10M to address our own needs.

What do you think ?

All the best,
Michel

**Comment by timlegge** [comment]
Someone at work mentioned that recently.  Its not limited on purpose.  I would say 10 or 20 M would be fine.  PR?

Tim

**Comment by timlegge** [comment]
Hi

Looking at 

sed /client_max_body_size/s/1m/50m/g  /etc/nginx/nginx.conf

TIm

**Comment by mmallejac** [comment]
Hi Tim,

Yes ! Will do... soon :-)

Le mar. 28 juil. 2020 à 03:07, Timothy Legge <notifications@github.com> a
écrit :

**Comment by mmallejac** [comment]
fixed by e0164df

**Comment by mmallejac** [comment]
Hi Tim,

Attaching files with size up to 50M works fine now,  but something is not good here.

Within a new Docker instance, attaching a PDF file is ok but opening it afterwards leads to an error : the browser receives a zero size file, while the file within `pub/web/topic` is correct.

I see that in the Nginx errorlog file, but not sure to understand it :

```
172.20.0.1 - - [03/Aug/2020:12:40:34 +0000] "GET /pub/Main/WebHome/8H_AC04.pdf?_t=1596458422 HTTP/1.1" 200 5 "http://localhost:8765/Main/WebHome" "Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:79.0) Gecko/20100101 Firefox/79.0"

==> foswiki-error.log <==
2020/08/03 12:40:34 [notice] 25#25: *63 "^/pub/(.*)$" matches "/pub/Main/WebHome/8H_AC04.pdf", client: 172.20.0.1, server: _, request: "GET /pub/Main/WebHome/8H_AC04.pdf?_t=1596458422 HTTP/1.1", host: "localhost:8765", referrer: "http://localhost:8765/Main/WebHome"
2020/08/03 12:40:34 [notice] 25#25: *63 rewritten data: "/bin/xsendfile/Main/WebHome/8H_AC04.pdf", args: "_t=1596458422", client: 172.20.0.1, server: _, request: "GET /pub/Main/WebHome/8H_AC04.pdf?_t=1596458422 HTTP/1.1", host: "localhost:8765", referrer: "http://localhost:8765/Main/WebHome"
```

I've also updated one of my instances, and on this one the behavior is slightly different : while opening the newly attached file, and instead of receiving an empty file, Foswiki answers : 

![image](https://user-images.githubusercontent.com/11666281/89183931-4eae1c00-d598-11ea-96fb-f439829fa915.png)

On this one, the complete URL is  : 
`https://(myWiki)/pub/Cuisine/MielPissenlit/8H_AC04.pdf?_t=1596449443`

If the attached file is a picture, it is correctly shown in the topic, the error occurs when trying to open the attached file from the topic.

Any idea ??

**Comment by timlegge** [comment]
Strange - it mentions  rewritten data: "/bin/xsendfile/Main/WebHome/8H_AC04.pdf" I wonder if the xsendfile change is related.  @MichaelDaum any thoughts?

**Comment by timlegge** [comment]
@mmallejac this version uses xfilesend do you have the XSendFileContrib enabled?

**Comment by mmallejac** [comment]
not sure, I wasn't aware of this
if this version uses xfilesend, shouldn't it be enabled by default within
the Dockerfile ?

Le mar. 4 août 2020 à 16:17, Timothy Legge <notifications@github.com> a
écrit :

**Comment by timlegge** [comment]
yes, but is this an existing wiki?  the volume would not have that change

**Comment by timlegge** [comment]
I removed xsendfilecontrib from my test volume and I get a 502 error.  Reinstalling it and the attachment worked ine (refresh the page though with F5 or <Ctrl>+<F5>)

**Comment by mmallejac** [comment]
Have you made a test with a fresh new install ?

Le mar. 4 août 2020 à 18:21, Timothy Legge <notifications@github.com> a
écrit :

**Comment by timlegge** [comment]
I have not.  Was your's a fresh install?

**Comment by mmallejac** [comment]
Yes : read my re-opening text of yesterday :-)
I've met 2 different situations, with the new image or with existing
volumes.

Le mar. 4 août 2020 à 21:05, Timothy Legge <notifications@github.com> a
écrit :

**Comment by timlegge** [comment]
I am tracking it down:

$Foswiki::cfg{XSendFileContrib}{Header} = 'X-Accel-Redirect';
$Foswiki::cfg{XSendFileContrib}{Location} = '/files';

Will have more info shortly and a patch

**Comment by timlegge** [comment]
The problem appears to be that I had the settings in the dockerfile before the Contrib was installed.  However, right now the auuto builds at github are failing.  If you want to try v1.18 at dockerhub it should be fine

**Comment by mmallejac** [comment]
Hi Tim,

Si I've juste tested the new image (5797d249fe98).

With exiting data : 
- First I still get the same error " Missing or illegal web name - You are trying to view in a web that does not exist"
- nothing worth mentioning in the `volumes/foswiki_www/working/logs` files (nothing about the download tentative)
- the nginx error log mention xsendfile : `2020/08/06 13:59:19 [notice] 21#21: *50 rewritten data: "/bin/xsendfile/Myweb/MyTopic/Myfile.pdf", ...`
- So I install the XSendFileContrib
- it looks better : the choosen attached file is sent to the browser, but size is 0
- no error in the logs (Foswiki and Nginx)

With fresh install : 
- works like a charm

So in my existing instances, I am probably missing another contrib or ?

**Comment by timlegge** [comment]
Did you add the following settings in configure for your existing wiki

{XSendFileContrib}{Header} = 'X-Accel-Redirect';
{XSendFileContrib}{Location} = '/files';

**Comment by mmallejac** [comment]
Hello Tim,

Just did it, and it now works fine with my existing instances ! thanks

Just one minor thing : when deleting an attachment, the "Deleting" message stays, I have then to refresh to have it gone.
But that happens only on existing instances.

**Comment by timlegge** [comment]
@mmallejac Weird about the refresh.  I take it the logs don't say much.  I will try to reproduce
